File: server/icecreamapp/src/main/java/com/ssafy/icecreamapp/model/dto/Member.java
package com.ssafy.icecreamapp.model.dto;

import com.ssafy.icecreamapp.model.dto.request.InitMember;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@Getter
@Setter
@ToString
@NoArgsConstructor
public class Member {
    int id;
    String name;
    String email;
    String password;
    int purchaseSum;
    int gender;
    int age;

    public Member(InitMember initMember) {
        this.name = initMember.getName();
        this.email = initMember.getEmail();
        this.password = initMember.getPassword();
        this.age = initMember.getAge();
        this.gender = initMember.getGender();
    }
}

File: android/app/src/main/java/com/ssafy/icepop/data/model/dto/Member.kt
package com.ssafy.icepop.data.model.dto

data class Member(
    val age: Int,
    val discountRate: Double,
    val email: String,
    val gender: Int,
    val level: String,
    val name: String,
    val nextLvRemain: Int
) {

    constructor() : this(
        age = 0,
        email = "",
        discountRate = 0.0,
        gender = 0,
        level = "",
        name = "",
        nextLvRemain = 0
    )

    // 보조 생성자
    constructor(age: Int, email: String, gender: Int) : this(
        age = age,
        email = email,
        discountRate = 0.0,
        gender = gender,
        level = "",
        name = "",
        nextLvRemain = 0
    )
}
